src/reynir/bintokenizer.py

```diff
@@ -1287,6 +1307,37 @@ def compatible(pn: PersonNameTuple, npn: PersonNameTuple) -> bool:
                 namespan = token.original or ""
                 patronym = False
                 while True:
+                    next_next_token = None
+                    # Deal with lowercase middle name abbreviations when auto uppercasing
+                    if auto_uppercase and next_token.txt in MIDDLE_NAME_ABBREVS:
+                        next_next_token = next(token_stream)
+                        following_tokens.append(next_next_token)
+
+                        # Check if next token is a period
+                        if next_next_token.punctuation == ".":
+                            # Join to middle name abbreviation
+                            next_token = next_token.concatenate(next_next_token)
+                            following_tokens.popleft()
```

Comment:
Are you sure that you want to pop from the left side, not from the right side (where you previously appended next_next_token)?

Reply:
Yes, I am really just using the deque as a queue ADT (as Python doesn't have a pure queue implementation in any library). next_next_token should always be at the front of the queue (or the left side of the deque) so popleft is intentional.

Reply:
On second thought, it is probably safer to use pop rather than popleft, good point!